examples/tree_1d_dgsem/elixir_shallowwater_multilayer_dam_break_es_dry.jl

```diff
@@ -0,0 +1,124 @@
+    # This default value can be changed within the constructor call depending on the simulation setup.
+    for i in reverse(eachlayer(equations))
+        if i == nlayers(equations)
+            H[i] = max(H[i], b + equations.threshold_limiter)
+        else
+            H[i] = max(H[i], H[i + 1] + equations.threshold_limiter)
+        end
+    end
+
+    # Initialize zero velocity
+    v = zero(H)
+
+    return prim2cons(SVector(H..., v..., b), equations)
+end
+
+initial_condition = initial_condition_dam_break
+
+###############################################################################
+# Get the DG approximation space
+
+volume_flux = (flux_ersing_etal, flux_nonconservative_ersing_etal)
+surface_flux = (FluxHydrostaticReconstruction(FluxPlusDissipation(flux_ersing_etal,
+                                                                  DissipationLocalLaxFriedrichs()),
+                                              hydrostatic_reconstruction_ersing_etal),
+                FluxHydrostaticReconstruction(flux_nonconservative_ersing_etal,
+                                              hydrostatic_reconstruction_ersing_etal))
+
+basis = LobattoLegendreBasis(3)
+
+indicator_sc = IndicatorHennemannGassnerShallowWater(equations, basis,
+                                                     alpha_max = 0.5,
+                                                     alpha_min = 0.001,
+                                                     alpha_smooth = true,
+                                                     variable = waterheight_pressure)
+volume_integral = VolumeIntegralShockCapturingHG(indicator_sc;
+                                                 volume_flux_dg = volume_flux,
+                                                 volume_flux_fv = surface_flux)
+
+solver = DGSEM(basis, surface_flux, volume_integral)
+
+###############################################################################
+# Get the TreeMesh and setup a non-periodic mesh
+
+coordinates_min = 0.0
+coordinates_max = 20.0
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level = 5,
+                n_cells_max = 10000,
+                periodicity = false)
+
+boundary_condition = boundary_condition_slip_wall
+
+# create the semidiscretization object
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition, solver,
+                                    boundary_conditions = boundary_condition)
+
+###############################################################################
+# ODE solvers
+
+tspan = (0.0, 1.0)
+ode = semidiscretize(semi, tspan)
+
+###############################################################################
+# Callbacks
+
+summary_callback = SummaryCallback()
+
+analysis_interval = 500
+analysis_callback = AnalysisCallback(semi, interval = analysis_interval,
+                                     save_analysis = false)
+
+stepsize_callback = StepsizeCallback(cfl = 0.2)
```

Comment:
It turned out to be necessary to lower the CFL for this test, but I think it's due to positivity reasons and not an issue with the wave speed estimate.
Since this test initializes a dam break over a dry domain, we need to set a smaller time step to ensure positivity.
Currently, there is no specific time step restriction implemented for this so I just manually set a small enough CFL number.

Reply:
That makes sense, the stable time step restriction can be quite restrictive for discontinuous flows with dry regions. I was just curious about the wave speed estimate.